### PR TITLE
Marvell: Migrate to multi console API

### PR DESCRIPTION
**Pull request by kostapr**
This patch fixes small bug in A3700 image encryption command and  introduces migration to the multi-console API for all supported Marvell platforms.

### CODE REVIEW

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

The idea with multi console is to remove the old console at some point. The reason why Arm platforms support both is that we need to test that everything keeps working until we can get rid of the old console model (that's the reason for the ``#if MULTI_CONSOLE_API`` in plat/arm).

In your case, you can remove the conditionals and just leave the multi console code (and make your code easier).

Also, you have a typo in your commit message: ``plat/marvell: Migrate to milti-console API``.

File: drivers/marvell/uart/a3700_console.S [hunk 1]
```diff
@@ -5,16 +5,26 @@
  * https://spdx.org/licenses
  */
 
+#include <arch.h>
 #include <asm_macros.S>
 #include <a3700_console.h>
```
Comment (antonio-nino-diaz-arm) [inline drivers/marvell/uart/a3700_console.S]: Actually, now that I'm taking a look at https://github.com/ARM-software/arm-trusted-firmware/pull/1683, you are missing ``#define USE_FINISH_CONSOLE_REG_2`` before the include of ``console_macros.S``.

Comment (kostapr) [reply drivers/marvell/uart/a3700_console.S]: After adding this define I do not see any output from TF-A BLx anymore.

Comment (kostapr) [reply drivers/marvell/uart/a3700_console.S]: Old output:
========
TIM-1.0
WTMI-devel-18.07.0-6050fd5
WTMI: system early-init
CPU VDD voltage default value: 1.155V
NOTICE:  Booting Trusted Firmware
NOTICE:  BL1: v2.0(release):v2.0-257-g036a759 (Marvell-devel-18.09.1)
NOTICE:  BL1: Built : 12:21:02, Nov 15 2018
NOTICE:  BL1: Booting BL2
NOTICE:  BL2: v2.0(release):v2.0-257-g036a759 (Marvell-devel-18.09.1)
NOTICE:  BL2: Built : 12:21:03, Nov 15 2018
NOTICE:  BL1: Booting BL31
NOTICE:  BL31: v2.0(release):v2.0-257-g036a759 (Marvell-devel-18.09.1)
NOTICE:  BL31: Built : 12:2

U-Boot 2018.03-devel-18.11.0-00290-g0e283d5-dirty (Nov 15 2018 - 12:20:01 +0200)

Model: Marvell Armada 3720 Community Board ESPRESSOBin
       CPU    @ 1000 [MHz]
       L2     @ 800 [MHz]
       TClock @ 200 [MHz]
       DDR    @ 800 [MHz]
DRAM:  1 GiB
Comphy chip #0:
Comphy-0: USB3          5 Gbps    
Comphy-1: PEX0          2.5 Gbps  
Comphy-2: SATA0         6 Gbps    
SATA link 0 timeout.

new output:
=========
TIM-1.0
WTMI-devel-18.07.0-6050fd5
WTMI: system early-init
CPU VDD voltage default value: 1.155V


U-Boot 2018.03-devel-18.11.0-00290-g0e283d5-dirty (Nov 15 2018 - 12:20:01 +0200)

Model: Marvell Armada 3720 Community Board ESPRESSOBin
       CPU    @ 1000 [MHz]
       L2     @ 800 [MHz]
       TClock @ 200 [MHz]
       DDR    @ 800 [MHz]
DRAM:  1 GiB
Comphy chip #0:
Comphy-0: USB3          5 Gbps    
Comphy-1: PEX0          2.5 Gbps  
Comphy-2: SATA0         6 Gbps    
SATA link 0 timeout.

Comment (kostapr) [reply drivers/marvell/uart/a3700_console.S]: Ignore my complain, I see what the problem - the driver registration macro has changed.

**Comment by kostapr** [comment]
Done

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
There are actual build errors with this PR in all non-a3700 marvell platforms (because of BLE):
```
/work/workspace/workspace/tf-worker/ble-marvell/ble_main.c: In function 'exec_ble_main':
/work/workspace/workspace/tf-worker/ble-marvell/ble_main.c:38:3: error: implicit declaration of function 'console_init' [-Werror=implicit-function-declaration]
   console_init(PLAT_MARVELL_BOOT_UART_BASE,
   ^~~~~~~~~~~~
```

**Comment by kostapr** [comment]
It surprises me. I did several builds with mainline TF-A source. I will re-check this ASAP

**Comment by kostapr** [comment]
Sounds like your build is using external BLE sources with BLE_DIR defined. This patch has updated the console initialization with the new calls, so no "console_init" is called there.
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 32)  /*
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 33)   * In some situations, like boot from UART, bootrom will
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 34)   * request to avoid printing to console. in that case don't
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 35)   * initialize the console and prints will be ignored
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 36)   */
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 37)  if ((bootrom_flags & BR_FLAG_SILENT) == 0)
1fb58fbf9 (Konstantin Porotchkin 2018-11-14 17:15:08 +0200 38)          marvell_console_boot_init();
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 39) 
37c4341b1 (Konstantin Porotchkin 2018-10-03 14:21:42 +0300 40)  NOTICE("Starting binary extension\n");

**Comment by antonio-nino-diaz-arm** [comment]
Yes, you are right, it's most likely a problem of our CI, we didn't remove the ``BLE_PATH`` from the build command line. I'll fix it and run the tests again.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Pull request merged**